tests/validate/common/includes/xcl2/xcl2.cpp

```diff
@@ -33,14 +37,21 @@ std::vector<cl::Device> get_devices(const std::string &vendor_name) {
     platform = platforms[i];
     OCL_CHECK(err, std::string platformName =
                        platform.getInfo<CL_PLATFORM_NAME>(&err));
-    if (platformName == vendor_name) {
+    if (!(platformName.compare(vendor_name))) {
```

Comment:
Is there any advantage to using std::string::compare over '=='?

Reply:
Technically both are same. It's just that .compare() is slightly faster "==" in debug mode.
https://stackoverflow.com/questions/9158894/differences-between-c-string-and-compare